pkg/cmd/pulumi/packagecmd/project_sdk_link.go

```diff
@@ -744,11 +726,12 @@ func SchemaFromSchemaSource(pctx *plugin.Context, packageSource string, args []s
 	if pluginSpec.Version != nil {
 		spec.Version = pluginSpec.Version.String()
 	}
-	if overrideName != "" {
-		spec.Name = overrideName
-	}
-	if overrideNamespace != "" {
-		spec.Namespace = overrideNamespace
+	if spec.Namespace == "" && strings.HasPrefix(pluginSpec.PluginDownloadURL, "git://") {
```

Comment:
Let's make a small function out of it and cover it with all the tests we can think of. I'm very bad at parsing regex and seeing if it works...